### PR TITLE
Move solutions into the root directory

### PR DESCRIPTION
**Pull request by agocke**
It's confusing for new users to have the Roslyn solutions nested
inside the src/ directory and there's really no good reason those where
there in the first place. This moves them to the root directory, except
for the Toolset.sln solution, which is only for building Roslyn in CI
scenarios and has been moved to the build/ directory.

@amcasey @tmeschter @gafter @dpoeschl @anyoneelsewhocares

### CODE REVIEW

File: CrossPlatform.sln [hunk 1]
```diff
@@ -677,14 +674,9 @@ Global
...
 		{6AA96934-D6B7-4CC8-990D-DB6B9DD56E34} = {FD0FAF5F-1DED-485C-99FA-84B97F3A8EEC}
-		{E195A63F-B5A4-4C5A-96BD-8E7ED6A181B7} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
-		{E3FDC65F-568D-4E2D-A093-5132FD3793B7} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
 		{909B656F-6095-4AC2-A5AB-C3F032315C45} = {FD0FAF5F-1DED-485C-99FA-84B97F3A8EEC}
-		{2E87FA96-50BB-4607-8676-46521599F998} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
 		{21B239D0-D144-430F-A394-C066D58EE267} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
-		{687DAFFD-9BD9-4331-96B7-483B941EDEAA} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
 		{57CA988D-F010-4BF2-9A2E-07D6DCD2FF2C} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
-		{E637AD92-8397-4337-A9CD-9F2570078E59} = {55A62CFA-1155-46F1-ADF3-BEEE51B58AB5}
 		{D0BC9BE7-24F6-40CA-8DC6-FCB93BD44B34} = {A41D1B99-F489-4C43-BBDF-96D61B19A6B9}
```
Comment (amcasey) [inline CrossPlatform.sln]: What were these?

**Comment by amcasey** [comment]
I'm assume we're okay with the fact that msbuild will no longer automatically select BuildAndTest.proj in the root directory?

**Comment by agocke** [comment]
@amcasey I think that's fine. We've gotten used to msbuild Roslyn.sln vs Test.exe and I don't think it's a big deal. To me it seems more convenient.

File: CrossPlatform.sln (hunk 1 above)
Comment (agocke) [inline CrossPlatform.sln]: I have no idea -- but when I closed and opened the solutions in VS, this is what happened to them.

**Pull request merged**